examples/escape-hatch/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+use futures::channel::oneshot;
+use handler::Connection;
+use libp2p::swarm::{
+    ConnectionDenied, ConnectionId, FromSwarm, SubstreamProtocol, THandler, THandlerOutEvent,
+    ToSwarm,
+};
+use libp2p::{swarm::NetworkBehaviour, tcp::tokio::TcpStream, PeerId, Stream, StreamProtocol};
+use libp2p_core::upgrade::ReadyUpgrade;
+use libp2p_core::{Endpoint, Multiaddr};
+use std::task::Context;
+use std::{collections::VecDeque, io, task::Poll, time::Duration};
+use std::sync::mpsc;
+
+mod handler;
+
+pub const PROTOCOL_NAME: StreamProtocol = StreamProtocol::new("/ipfs/ping/1.0.0");
+
+#[derive(Debug)]
+pub enum Error {
+    NotConnected(PeerId),
+    UnsupportedProtocol,
+    Io(io::Error),
+}
+
+/// Event generated by the `Ping` network behaviour.
+#[derive(Debug)]
+pub struct Event {
+    /// The peer ID of the remote.
+    pub peer: PeerId,
+    /// The connection the ping was executed on.
+    pub connection: ConnectionId,
+    /// The result of an inbound or outbound ping.
+    pub result: Result<Duration, Error>,
+}
+
+/// A behaviour that manages requests to open new streams which are directly handed to the user.
+pub struct Behaviour {
+    /// Config
+    config: Config,
+    /// Queue of events to yield to the swarm.
+    events: VecDeque<Event>,
+    /// Protocol
+    protocol: StreamProtocol,
+}
+
+#[derive(Debug, Clone)]
+pub struct Config {
+    /// The timeout of an outbound ping.
+    timeout: Duration,
+    /// The duration between outbound pings.
+    interval: Duration,
+}
+
+struct StreamMessage {
+    peer_id: PeerId, // Assuming PeerId is a type representing a peer
+    response_channel: oneshot::Sender<Result<Stream, Error>>, // For sending back the result
+}
+
+pub struct IncomingStreams {
+    /// Queue of events to yield to the swarm.
+    events: VecDeque<Event>,
+}
+
+/// A control acts as a "remote" that allows users to request streams without interacting with the `Swarm` directly.
+
+pub struct Control {
+    sender: mpsc::Sender<StreamMessage>,
+    receiver: mpsc::Receiver<StreamMessage>
+}
+
+impl IncomingStreams {
+    pub async fn next(&mut self) -> (PeerId, Stream) {
+        if let Some(e) = self.events.pop_back() {
+        } else {
+        }
+
+        todo!()
```

Comment:
Similar to `Control::open_stream`, this needs to listen on an `mpsc::Receiver` where the `mpsc::Sender` lives in `Behaviour`. Any inbound streams from the `ConnectionHandler` need to be sent this one.